folly/system/ThreadId.h

```diff
@@ -84,8 +88,10 @@ inline uint64_t getOSThreadID() {
   return tid;
 #elif _WIN32
   return uint64_t(GetCurrentThreadId());
-#else
+#elif __linux__
   return uint64_t(syscall(FOLLY_SYS_gettid));
+#else
```

Comment:
Is this for BSD? Can we guard it explicitly?

Reply:
phread_np isn't really detectable by OS alone, that's why I originally had it explicitly guarded for if that header was present. should I change it back?

Reply:
Guard it with the same check `__has_include(<pthread_np.h>)`?

Reply:
done. I added an error as the last case, following the pattern in Portability.h.